private/waiter/waiter.go

```diff
@@ -61,20 +61,56 @@ func (w *Waiter) Wait() error {
 					break
 				}
 				result = true
-				for _, val := range vals {
-					if !awsutil.DeepEqual(val, a.Expected) {
-						result = false
-						break
+				if exp, ok := a.Expected.(string); ok {
```

Comment:
`AutoScaling`'s waiters are a good example of bool expected types. e.g. [`WaitUntilGroupInService`](https://github.com/aws/aws-sdk-go/blob/master/service/autoscaling/waiters.go#L38). In generally only `bool`, `string`, and `int` values are used as `Expected`.

If the fields were split out they would be `ExpectedString`, `ExpectedInt`, `ExpectedBool`. This would also grow if the `Expected` for each additional type that were ever added such as float, slice, and maps. With that said, this would be a breaking change to the SDK because the `waiter.WaitAcceptor` type is an exported type. User's most likely would have no reason to use the type or its fields directly, but changing the fields would break them if they did.

There is nothing wrong with updating this section to use non-reflection for for the known cases int, string, bool, and falling back to the `awsutil.DeepEquals` for all other cases. In addition to the change should not be hardcodeed and duplicated inside of the waiter code, but implemented in a standalone utility function that is testable. In addition this should be discussed in its own Issue, with its own PR.